### PR TITLE
Move Indy NodeJS wrapper codebase here

### PR DESCRIPTION
**Pull request by ajile-in**
To start with I am moving the Indy NodeJS wrapper as it is here, with no code changes at all except few Readme changes.

### CODE REVIEW

**Review by smithbk — COMMENTED** [review]
Jadhavajay, I understand that you are just getting started, but I'd like to understand the goal of this SDK before I review.  This is probably better to discuss on a call, but I assume the goal is to implement a higher level agent API.

Since Aries is supposed to allow other implementations other than indy, it would seem reasonable that we at least start by deciding on a directory structure which would allow other implementations to be plugged in.

For example, perhaps something similar to:
```
-- api (The Aries agent API - to be defined)
    -- spi (The service provider API which is used by the Aries agent APIs - i.e. an abstraction of the indy APIs)
        -- indy (the current indy code goes here)
            -- test (indy test code here)
        -- another implementation of the SPI in the future
-- test (test for the higher-level agent APIs)
```

File: src/IndyError.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  Error.captureStackTrace(this, this.constructor)
+  this.name = this.constructor.name
+  if (errors.hasOwnProperty(err)) {
+    this.message = errors[err]
+    this.indyCode = err
+    this.indyName = errors[err]
+    try {
+      this.indyCurrentErrorJson = capi.getCurrentError()
+      var details = JSON.parse(this.indyCurrentErrorJson)
+      if (typeof details.message === 'string') {
```
Comment (matt-raffel-kiva) [inline src/IndyError.js]: var should be replaced with const to avoid potential conflicts in global namespace

File: src/IndyError.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+var util = require('util')
```
Comment (matt-raffel-kiva) [inline src/IndyError.js]: every line should be terminated with semicolon.  this is an industry standard and is enforced with linters.

File: src/IndyError.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      this.indyCurrentErrorJson = capi.getCurrentError()
+      var details = JSON.parse(this.indyCurrentErrorJson)
+      if (typeof details.message === 'string') {
+        this.indyMessage = details.message
+      }
+      if (typeof details.backtrace === 'string') {
+        this.indyBacktrace = details.backtrace
+      }
+    } catch (e) {
+    }
```
Comment (matt-raffel-kiva) [inline src/IndyError.js]: we need to do something here.  just eating the exception will make it very difficult to debug when a problem happens.

File: src/wrapIndyCallback.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+var IndyError = require('./IndyError')
+
```
Comment (matt-raffel-kiva) [inline src/wrapIndyCallback.js]: all lines should be terminated with semicolon.

File: src/wrapIndyCallback.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+function wrapIndyCallback (cb, mapResponse) {
+  var promise
+  if (!cb) {
```
Comment (matt-raffel-kiva) [inline src/wrapIndyCallback.js]: var should be replaced with let

File: test/anoncreds.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
+var test = require('ava')
```
Comment (matt-raffel-kiva) [inline test/anoncreds.js]: every statement should be terminated with semicolon

File: test/anoncreds.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+var indy = require('../')
+var cuid = require('cuid')
+var initTestPool = require('./helpers/initTestPool')
+var tempy = require('tempy')
+
+test('anoncreds', async function (t) {
+  var pool = await initTestPool()
```
Comment (matt-raffel-kiva) [inline test/anoncreds.js]: local variables need to be declared with const or let.

File: test/blob.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+var test = require('ava')
```
Comment (matt-raffel-kiva) [inline test/blob.js]: please see my comments for the previous files.  same for this one.

File: test/cache.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
+var test = require('ava')
```
Comment (matt-raffel-kiva) [inline test/cache.js]: please see my comments for the previous files.  same for this one.

File: test/crypto.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
+var test = require('ava')
```
Comment (matt-raffel-kiva) [inline test/crypto.js]: please see my comments for the previous files.  same for this one.

File: test/did.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
+var test = require('ava')
```
Comment (matt-raffel-kiva) [inline test/did.js]: please see my comments for the previous files.  same for this one.

**Review by matt-raffel-kiva — CHANGES_REQUESTED** [review]
For all js files:  terminate each statement with semicolon.  replace var with const and let in functions.

**Comment by ajile-in** [comment]
Thanks @matt-raffel-kiva for your comments on this PR.
This PR is just a code move from previous Indy Node.js wrapper repository.

We (@smithbk & me) discussed last week to have a Zoom call together and discuss the roadmap of how to take this further with Aries Framework idea in mind. We also discussed about using TypeScript instead of ES.

I will catch up with you on RocketChat and will schedule a Zoom call.

**Comment by esplinr** [comment]
I was surprised to see this PR closed. I agree with @smithbk that we should define how to be ledger-independent, but I don't think that should prevent taking the existing Indy code as the foundation for an aries-js-wrapper and then iterating. It's hard for interested contributors to know how to collaborate when the repository is completely empty. A simple README.md explaining the current state of the project and current efforts would also help a lot.

**Comment by matt-raffel-kiva** [comment]
I don’t have maintainer rights on that repo.  If I could get that, then I could help out more.

**Comment by ryjones** [comment]
I apologize. I did a force-push to fix a DCO issue, because I didn't see there was a PR in flight. This is my fault.

**Comment by ryjones** [comment]
@esplinr I can't re-open the PR b/c master changed under it.

**Comment by esplinr** [comment]
Thanks for the explanation @ryjones . That makes sense. I misinterpreted what was going on based on the comments, but now I see that @smithbk was only prompting a helpful discussion and a decision hadn't yet been made.

@matt-raffel-kiva: I think @jadhavajay can work with @ryjones to grant those rights.

**Pull request closed without merging**